NS-Proto-Appendix.md

```diff
@@ -8,41 +8,27 @@ Converts shallowly an object to a record. If the object has a non-const value, a
 
 Checks whether the parameter is either a Record primitive or Record wrapper.
 
-## `Record.assign(...args: Record[]) -> Record`
-
-Merges all records passed as arguments into one **new** record returned by the function. The latest argument will override arguments before when merging.
-
-## `Record.entries(record: Record) -> Tuple`
-
-Returns all of the entries in a Record as a Tuple of 2-values Tuples (key, value) in the same order as `Record.keys()`.
-
 ## `Record.fromEntries(iterator: Iterator): Record`
 
-Takes an iterator of 2-element array-like values and creates a record out of it. If the iterator returns something other than an array-like tuple, a TypeError will be raised.
-
-## `Record.keys(record: Record) -> Tuple`
-
-Returns a Tuple containing the values serving as keys. Order will be the one defined in the main proposal document.
-
-## `Record.toString(record: Record) -> String`
-
-To be defined - it can actually show the full value.
+Similar to `Object.fromEntries`, but created a new `Record`.
 
-## `Record.values(record: Record) -> Tuple`
+## `Object` namespace functions working with `Record`
 
-Returns all of the values in a Record in the same order as the associated keys in `Record.keys()`.
+- `Object.is()`
+- `Object.entries()`
+- `Object.keys()`
+- `Object.values()`
+- `Object.assign()` - **Caveat**: The first argument cannot be a record or a tuple, because they are immutable
 
-## `Object` namespace functions not available in `Record`
+## `Object` namespace functions not working with `Record`
 
 - `Object.create()` - `Record` has no prototype so `Record.create()` would not have any meaning.
 - `Object.defineProperties()`, `Object.defineProperty()` - A `Record` is non-observable, not writable by nature
```

Comment:
This function will "work" on a Record wrapper but not do anything really useful (since they act like they're frozen). It'll throw on Record and Tuple values, which is probably fine to leave as is.

An alternative for Object.defineProperty would be to make it call ToObject on Records and Tuples, and do [[DefineProperty]] on that. I don't have much of an opinion about whether we do this now; it maybe more of an attractive change if there is a more general future value types proposal.

Reply:
It won’t throw if you try to define the same values.

Reply:
It will, because `Object.defineProperty` throws on non-objects, in its first step, before we get to the [[DefineProperty]] behavior for Record wrappers. So, that's something that we can go either way on relaxing IMO. We can also go either way on relaxing it for `Reflect`. Ultimately, you can always access this functionality by casting to Object explicitly, and also the functionality is not very useful.

Reply:
ah, my mistake for the primitive form. seems reasonable to keep the exception for the primitive if there’s no useful defining to be done in it.